Add migration guide for react hooks for ably-js v2

VeskeR opened the pull request:
Resolves #1684

VeskeR commented on docs/react-migration-guide.md:
```diff
@@ -1,5 +1,207 @@
 # React hooks upgrade / migration guide
 
+## Version ?? to ??
```
Previous iterations of this migration guide used versions used by separate react-hooks library, which was not part of ably-js repo (see line https://github.com/ably/ably-js/pull/1698/files#diff-f8ce6205300ecf411be3e91af62c0ecbaf88971e7cd082fc9193ef20cdea32eeR205), so previous guide was for `Version 2.x to 3.x`.

What do we want write now here?
React Hooks are now part of ably-js and follow its versioning.

owenpearson replied:
the "2.x to 3.x" thing shouldn't have been there in the first place (probably my bad) - we were originally making some breaking changes as part of an `@ably-labs/react-hooks` v3 major release but instead decided to merge the hooks into the core `ably` package so there never was a 3.x release in the end.

I think we should change the header of the previous migration guide to "`@ably-labs/react-hooks` to 1.x" and then the new migration guide can simply be labelled "1.x to 2.x"

VeskeR replied:
how about "`@ably-labs/react-hooks` to `ably-js 1.x`"? so it's a bit more descriptive that we moved to `ably-js` repo

owenpearson replied:
sounds good to me, but i would go for `ably` rather than `ably-js` since that's the name of the npm package

VeskeR replied:
yea, makes sence! will do `ably`

VeskeR replied:
changed in https://github.com/ably/ably-js/commit/1c5d80da35f5ca560afb988b4df7cfd7ce23d6ef

owenpearson commented on docs/react-migration-guide.md:
```diff
@@ -1,6 +1,208 @@
+
+### Renaming `id` to `ablyId`
+
+All instances of the `id` field, which were used to specify the `AblyProvider` component and the underlying Ably client to use, have been renamed to `ablyId`.
```
I think it would be nice to make it clear that this was an optional feature, i'd bet the vast majority of users aren't using more than one provider so this section might be a bit confusing to them
```suggestion
All instances of the `id` field, which optionally were used to specify the `AblyProvider` component and the underlying Ably client to use, have been renamed to `ablyId`.
```

VeskeR replied:
True, will also change the header to `Renaming optional "id" field to "ablyId"`

owenpearson commented on docs/react-migration-guide.md:
```diff
@@ -1,6 +1,208 @@
+// in a child component:
+const { channel } = useChannel({ channelName: 'your-channel-name' }, (message) => {
+  console.log(message);
+});
```
```suggestion
});

// use the useState setter to change the options provided to the `ChannelProvider` component
setOptions({ params: { rewind: '1' } });
```

owenpearson commented on docs/react-migration-guide.md:
```diff
@@ -1,6 +1,208 @@
+
+### ChannelProvider
+
+In previous versions, you were able to provide channel options as a parameter in the `useChannel`/`usePresence` hooks. This often led to errors when using hooks with the same channel name but different options, or when attempting to dynamically change options for a channel.
```
i think slightly softer language might make this sound less scary 😅 
```suggestion
In previous versions, you were able to provide channel options as a parameter in the `useChannel`/`usePresence` hooks. This could, in some cases, lead to errors when using hooks with the same channel name but different options, or when attempting to dynamically change options for a channel.
```

VeskeR replied:
Agree 😅

owenpearson reviewed (approved):
Looks good! A few (optional) suggestions

VeskeR commented on docs/react-migration-guide.md:
````diff
@@ -1,6 +1,208 @@
+});
+```
+
+### Renaming `id` to `ablyId`
````
```suggestion
### Renaming optional `id` field to `ablyId`
```

jamienewcomb reviewed (approved):
Approving from a readability perspective

VeskeR commented:
Moved migration guide for React hooks v1 -> v2 to a separate file to conform to the structure introduced in  https://github.com/ably/ably-js/pull/1670/commits/9ee56819cdb4f59cb21e601cc2d37e2f71e92405 (part of https://github.com/ably/ably-js/pull/1670 PR).

Rebased this PR on top of https://github.com/ably/ably-js/pull/1670

ttypic reviewed (commented):
Looks great!
